src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  actions.forEach((action) => {
+    const previousState =
+      states.length > 0 ? states[states.length - 1] : { ...state };
+    let newState = { ...previousState };
```

Comment:
[NAMING] - Instead of naming the object `newState`, a more descriptive name like `stateCopy` would better convey the purpose of the object. Consider renaming it to follow the good example provided in the checklist.